### PR TITLE
Underflow fix

### PR DESCRIPTION
**Pull request by AntoineRondelet**
Quick fix for #170

### CODE REVIEW

File: src/circuits/joinsplit.tcc [hunk 1]
```diff
@@ -266,15 +271,21 @@ public:
...
+                        ),
                     output_commitments[j]->bits.rend());
             }
 
             // Initialize the unpacked input corresponding to the h_sig
             unpacked_inputs[NumOutputs + NumInputs].insert(
                 unpacked_inputs[NumOutputs + NumInputs].end(),
                 h_sig->bits.rbegin() +
-                    (HashT::get_digest_len() - FieldT::capacity()),
+                        safe_subtraction(
```
Comment (dtebbs) [inline src/circuits/joinsplit.tcc]: Looks like it's indented further than the old code...
I've just notived that I missed the code formatting checks in the github actions, hence the CI doesn't pick this up ... fixing that now.

File: src/circuits/joinsplit.tcc [hunk 2]
```diff
@@ -331,7 +352,10 @@ public:
                                     .end(),
                             input_nullifiers[NumInputs - i - 1]->bits.rbegin(),
                             input_nullifiers[NumInputs - i - 1]->bits.rbegin() +
-                                (HashT::get_digest_len() - FieldT::capacity()));
+                                safe_subtraction(HashT::get_digest_len(),
```
Comment (dtebbs) [inline src/circuits/joinsplit.tcc]: Probably another formatting issue.

File: src/circuits/joinsplit.tcc (hunk 1 above)
Comment (dtebbs) [reply src/circuits/joinsplit.tcc]: Reinstated formatting checks in #172 
Sorry about that.

Comment (AntoineRondelet) [reply src/circuits/joinsplit.tcc]: Thanks @dtebbs will wait for the checks to pass on #172 and merge it. Will fix the formatting issues afterwards

File: src/circuits/joinsplit.tcc [hunk 3]
```diff
@@ -254,7 +256,10 @@ public:
                 unpacked_inputs[i].insert(
                     unpacked_inputs[i].end(),
                     input_nullifiers[i]->bits.rbegin() +
-                        (HashT::get_digest_len() - FieldT::capacity()),
+                        safe_subtraction(
```
Comment (dtebbs) [inline src/circuits/joinsplit.tcc]: Might make sense to factor some of these out into a `const size_t` and save some lines.

File: src/circuits/joinsplit.tcc (hunk 1 above)
Comment (dtebbs) [reply src/circuits/joinsplit.tcc]: Thanks for merging #172.  Hopefully th CI will show errors once this is rebased onto `develop`, but if not let me know and I'll invesstigate.

File: src/circuits/joinsplit.tcc (hunk 3 above)
Comment (AntoineRondelet) [reply src/circuits/joinsplit.tcc]: Yes indeed, and more importantly you save useless function calls all the way down ;)
Changed in last commit

**Pull request merged**